### PR TITLE
Completed persistentParrots port

### PR DESCRIPTION
**Pull request by DeadlyMC**

### CODE REVIEW

File: src/main/java/carpet/mixins/PlayerEntity_parrotMixin.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        if (!this.world.isClient && ((parrots_will_drop && this.fallDistance > 0.5F) || this.isInWater() || (parrots_will_drop && this.hasVehicle())) || this.abilities.flying)
+        {
+            this.dropShoulderEntities();
+        }
+    }
+    
+    @Redirect(method = "damage", at = @At(value = "INVOKE",
+              target = "Lnet/minecraft/entity/player/PlayerEntity;dropShoulderEntities()V"))
+    private void cancelDropShoulderEntities2(PlayerEntity playerEntity)
+    {
```
Comment (gnembon) [inline src/main/java/carpet/mixins/PlayerEntity_parrotMixin.java]: why not to just use one @Redirect in this case, and just redirect dropShoulderEntities.

File: src/main/java/carpet/mixins/PlayerEntity_parrotMixin.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    }
+    
+    @Inject(method = "tickMovement", at = @At(value = "INVOKE", shift = At.Shift.AFTER, ordinal = 1,
+            target = "Lnet/minecraft/entity/player/PlayerEntity;updateShoulderEntity(Lnet/minecraft/nbt/CompoundTag;)V"))
+    private void onTickMovement(CallbackInfo ci)
+    {
+        boolean parrots_will_drop = !CarpetSettings.getBool("persistentParrots") || this.abilities.invulnerable;
+        if (!this.world.isClient && ((parrots_will_drop && this.fallDistance > 0.5F) || this.isInWater() || (parrots_will_drop && this.hasVehicle())) || this.abilities.flying)
+        {
+            this.dropShoulderEntities();
```
Comment (gnembon) [inline src/main/java/carpet/mixins/PlayerEntity_parrotMixin.java]: also in this case, one redirect on dropShoulderEntities with a few extra checks should make a simpler mixin

File: src/main/java/carpet/mixins/PlayerEntity_parrotMixin.java (hunk 1 above)
Comment (DeadlyMC) [reply src/main/java/carpet/mixins/PlayerEntity_parrotMixin.java]: I can't make use of **float_1** with the @ Redirect

Comment (gnembon) [reply src/main/java/carpet/mixins/PlayerEntity_parrotMixin.java]: I didn't notice that. That's correct. But the first one could be replaced with one redirect I guess.

**Pull request merged**